forui/lib/src/widgets/text_field/text_field_style.dart

```diff
@@ -0,0 +1,234 @@
+  final Color cursor;
+
+  /// The padding surrounding this text field's content. Defaults to
+  /// `const EdgeInsets.symmetric(horizontal: 16, vertical: 7)`.
+  final EdgeInsets contentPadding;
+
+  /// Configures padding to edges surrounding a [Scrollable] when this text field scrolls into view.
+  ///
+  /// Defaults to `EdgeInsets.all(20.0)`.
+  ///
+  /// When this widget receives focus and is not completely visible (for example scrolled partially off the screen or
+  /// overlapped by the keyboard) then it will attempt to make itself visible by scrolling a surrounding [Scrollable],
+  /// if one is present. This value controls how far from the edges of a [Scrollable] the TextField will be positioned
+  /// after the scroll.
+  final EdgeInsets scrollPadding;
+
+  /// The style when this text field is enabled.
+  final FTextFieldStateStyle enabled;
+
+  /// The style when this text field is enabled.
+  final FTextFieldStateStyle disabled;
+
+  /// The style when this text field has an error.
+  final FTextFieldStateStyle error;
+
+  /// The style when this text field is focused.
+  final FTextFieldStateStyle focused;
+
+  /// The style when this text field is focused and has an error.
+  final FTextFieldStateStyle focusedError;
+
+  /// Creates a [FTextFieldStyle].
+  FTextFieldStyle({
+    required this.label,
+    required this.keyboardAppearance,
+    required this.enabled,
+    required this.disabled,
+    required this.error,
+    required this.focused,
+    required this.focusedError,
+    this.cursor = CupertinoColors.activeBlue,
+    this.contentPadding = const EdgeInsets.symmetric(horizontal: 15, vertical: 7.5),
+    this.scrollPadding = const EdgeInsets.all(20.0),
+  });
+
+  /// Creates a [FTextFieldStyle] that inherits its properties.
+  FTextFieldStyle.inherit({
+    required FColorScheme colorScheme,
+    required FFont font,
+    required FStyle style,
+  }):
+    label = TextStyle(
+      color: colorScheme.primary,
+      fontSize: font.sm,
+      fontWeight: FontWeight.w600,
+    ),
+    keyboardAppearance = colorScheme.brightness,
+    cursor = CupertinoColors.activeBlue,
+    contentPadding = const EdgeInsets.symmetric(horizontal: 15, vertical: 7.5),
+    scrollPadding = const EdgeInsets.all(20.0),
+    enabled = FTextFieldStateStyle.inherit(
+      contentColor: colorScheme.primary,
+      hintColor: colorScheme.mutedForeground,
+      borderColor: colorScheme.border,
+      font: font,
+      style: style,
+    ),
+    disabled = FTextFieldStateStyle.inherit(
+      contentColor: colorScheme.primary,
+      hintColor: colorScheme.border.withOpacity(0.5),
+      borderColor: colorScheme.border.withOpacity(0.5),
+      font: font,
+      style: style,
+    ),
+    error = FTextFieldStateStyle.inherit( // TODO: add error colors
+      contentColor: colorScheme.primary,
+      hintColor: colorScheme.mutedForeground,
+      borderColor: colorScheme.border,
+      font: font,
+      style: style,
+    ),
```

Comment:
I think we have generic error colors currently in the theme. We could potentially add those in?